os/migrating-to-clcs.md

```diff
@@ -0,0 +1,377 @@
+# Migrating from Cloud Configs to Container Linux Configs
+
+Historically, the recommended way to provision a Container Linux machine was with a cloud config. This was a YAML file specifying things like systemd units to run, users that should exist, and files that should be written. This file would be given to a Container Linux machine, and saved on disk. Then a utility called coreos-cloudinit running in a systemd unit would read this file, look at the system state, and make necessary changes on every boot.
+
+Going forward, a new method of provisioning via Container Linux Configs is preferred.
+
+This document details how to convert an existing cloud config into a Container Linux Config. Once a Container Linux Config has been written, it is given to the Config Transpiler which will convert it into an Ignition Config. This Ignition Config can then be provided to a booting machine. For more information on this process, take a look at the [Config Transpiler getting started doc][ct-getting-started].
+
+The etcd and flannel examples shown in this document will use dynamic data in the Container Linux Config (anything looking like this: `{PRIVATE_IPV4}`). Not all types of dynamic data are supported on all cloud providers, and if the machine is not on a cloud provider this feature cannot be used. Please see [here][dynamic-data] for more information.
+
+To see all supported options available in a Container Linux Config, please look at the [Container Linux Config schema][ct-config].
+
+## coreos
+
+### etcd2
+
+In a cloud config, etcd version 2 can be enabled and configured by using the `coreos.etcd2.*` section. As an example of this:
+
+```cloud-config
+#cloud-config
+
+coreos:
+  etcd2:
+    discovery:                   "https://discovery.etcd.io/<token>"
+    advertise-client-urls:       "http://$public_ipv4:2379"
+    initial-advertise-peer-urls: "http://$private_ipv4:2380"
+    listen-client-urls:          "http://0.0.0.0:2379,http://0.0.0.0:4001"
+    listen-peer-urls:            "http://$private_ipv4:2380,http://$private_ipv4:7001"
+```
+
+In a Container Linux Config etcd can also be configured, but a Container Linux Config will use the etcd-member service on a Container Linux machine instead of the etcd2 service like coreos-cloudinit would.  The etcd-member service will download a version of etcd of the user's choosing and run it. This means that in a Container Linux Config both etcd v2 and v3 can be configured.
```

Comment:
First sentence: "Etcd can be configured in a more general way with a Container Linux Config."
Second sentence: "This CL Config will use the `etcd-member.service` systemd unit rather than the etcd2 service understood by cloud-config and coreos-cloudinit."